### ISSUE TITLE
While playing errors appear but everything works as expected

### ISSUE THREAD

**Issue by enaut** [Rusketh/ComparableHoppers2#22]
This form is for submitting issues with Comparable hoppers 2 for Spigot, please make sure to fill at all questions properlly as failing to do so will prevent us from helping replicate/fixing your issue.
This issue tracker is for Comparbale Hopper 2 Buyers only, if you have an issue with the plugin on a server but you are not the server owner please have the server owner or the plugins buyer fill out this form.

1) Q: What is your name on spigot?
   A: enaut

2) Q: What is your Spigot build?
   A: This is a final build for 1.12.2

3.1) Q: What is your product?
   E.G: CH2 Basic, CH2 Paid, CH2 Spawners...
   A: CH2 Basic,
   
3.2) Q: What is your Product version?
   E.G CH 2.3.1 for MC 1.12.x
   A: Enabling ComparableHoppers2 vBasic 2.4.2 custom build from #21

4) Q: Do you get any errors in console?
   A: Logs are not logged to file but to systemd - the amount of exceptions is huge a full log would be Megabytes... here is the error:

```
 [22:15:59 ERROR]: Could not pass event PlayerPickupItemEvent to ComparableHoppers2 vBasic 2.4.2
 org.bukkit.event.EventException: null
         at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.EntityItem.d(EntityItem.java:331) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.EntityHuman.c(EntityHuman.java:418) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.EntityHuman.n(EntityHuman.java:393) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.EntityLiving.B_(EntityLiving.java:1946) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.EntityHuman.B_(EntityHuman.java:145) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.EntityPlayer.playerTick(EntityPlayer.java:313) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.PlayerConnection.e(PlayerConnection.java:139) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.NetworkManager.a(NetworkManager.java:233) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.ServerConnection.c(ServerConnection.java:140) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:845) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
         at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
 Caused by: java.lang.NullPointerException
         at rusketh.com.github.hoppersbasic.helpers.NBTHelper.hasNBTKey(NBTHelper.java:196) ~[?:?]
         at rusketh.com.github.hoppersbasic.helpers.NBTHelper.getNBTBool(NBTHelper.java:201) ~[?:?]
         at rusketh.com.github.hoppersbasic.listeners.InventoryListener.PreventPickUp(InventoryListener.java:182) ~[?:?]
         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_181]
         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
         at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
         at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot.jar:git-Spigot-2cf50f0-2b93d83]
         ... 17 more
```

5) Q: Have you read the full OP for the resource (https://www.spigotmc.org/resources/comparable-hoppers-2.45058/
   If not you should probably do so and double check you have set everything up correctly.
   A: Yes
   
6) Q: Please describe the issue you are having.
   E.G When trying to craft a comparable upgrade a console error happens (as mentioned in Q4).
   A: Errors in the log, but not in the gameplay

**Comment by enaut** [comment]
Oh wait might be already fixed in new release?

**Comment by enaut** [comment]
No even  with fully updated version 2.5.5 (2.5.4 - I think something with the versioning is not right too I definately downloaded 2.5.5):
```
[18:35:52 ERROR]: Could not pass event PlayerPickupItemEvent to ComparableHoppers2 vBasic 2.5.4
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.EntityItem.d(EntityItem.java:331) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.EntityHuman.c(EntityHuman.java:418) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.EntityHuman.n(EntityHuman.java:393) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.EntityLiving.B_(EntityLiving.java:1946) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.EntityHuman.B_(EntityHuman.java:145) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.EntityPlayer.playerTick(EntityPlayer.java:313) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.PlayerConnection.e(PlayerConnection.java:139) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.NetworkManager.a(NetworkManager.java:233) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.ServerConnection.c(ServerConnection.java:140) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:845) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
Caused by: java.lang.NullPointerException
        at rusketh.com.github.hoppersbasic.helpers.NBTHelper.hasNBTKey(NBTHelper.java:196) ~[?:?]
        at rusketh.com.github.hoppersbasic.helpers.NBTHelper.getNBTBool(NBTHelper.java:201) ~[?:?]
        at rusketh.com.github.hoppersbasic.listeners.InventoryListener.PreventPickUp(InventoryListener.java:182) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_181]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot.jar:git-Spigot-2cf50f0-2b93d83]
        ... 17 more
```

**Comment by Rusketh** [comment]
This sounds like an issue with the NBT-API, the 1.8 & 1.8 snapshots are broken, we confirmed this with the author of the nbt api who has now released the 1.8.1 snapshot (I am talking NBT-API plugin versions not MC version here).

Please update your nbt api and confirm results thank you.

**Comment by enaut** [comment]
Thanks for the support... With the newer NBT it works!

I purchased it now even though I do not use the premium features...

I'd really like the thing to go open source though!
